javalin/src/main/java/io/javalin/http/util/MultipartUtil.kt

```diff
@@ -18,13 +19,18 @@ object MultipartUtil {
 
     private val defaultConfig = MultipartConfigElement(System.getProperty("java.io.tmpdir"), -1, -1, 1)
 
+    // TODO: This should be fetched from config instead
+    @Deprecated("Use `config.jetty.multipartConfig = new MultipartConfig()` instead")
     var preUploadFunction: (HttpServletRequest) -> Unit = {
         if (it.getAttribute(MULTIPART_CONFIG_ATTRIBUTE) == null) {
             it.setAttribute(MULTIPART_CONFIG_ATTRIBUTE, defaultConfig)
         }
     }
 
     private inline fun <R> HttpServletRequest.processParts(body: (Sequence<Part>, Int) -> R): R {
+        if ((this as JavalinServletRequest).isInputStreamRead()) {
+            throw IllegalStateException("Request body has already been consumed. You can't get multipart parts after reading the request body.")
```

Comment:
Can we cover this with our own exception, like e.g. BodyAlreadyReadException

Reply:
Yes, thank you.